source/common.c

```diff
@@ -158,10 +172,25 @@ void *aws_mem_acquire_many(struct aws_allocator *allocator, size_t count, ...) {
 #undef AWS_ALIGN_ROUND_UP
 
 void aws_mem_release(struct aws_allocator *allocator, void *ptr) {
-    allocator->mem_release(allocator, ptr);
+    AWS_PRECONDITION(allocator != NULL);
+    AWS_PRECONDITION(allocator->mem_release != NULL);
+
+    if (ptr != NULL) {
+        allocator->mem_release(allocator, ptr);
+    }
 }
 
 int aws_mem_realloc(struct aws_allocator *allocator, void **ptr, size_t oldsize, size_t newsize) {
+    AWS_PRECONDITION(allocator != NULL);
+    AWS_PRECONDITION(allocator->mem_realloc || allocator->mem_acquire);
+    AWS_PRECONDITION(allocator->mem_release);
+
+    /* Protect against https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations */
+    if (newsize == 0) {
```

Comment:
This one's legit, since people are lazy idiots, but maybe still wrap this in `AWS_UNLIKELY`.